profile/src/v100/entity/abstract_entity_type.rs

```diff
@@ -0,0 +1,55 @@
+use crate::prelude::*;
+use radix_engine_common::types::EntityType as EngineEntityType;
+
+/// Type of a wallet Radix Entity - Account or Identity (used by Personas).
+///
+/// CAP26 uses this type to create separate key spaces for Accounts and Identities
+#[derive(
+    Serialize,
+    Deserialize,
+    FromRepr,
+    Clone,
+    Copy,
+    Debug,
+    PartialEq,
+    Eq,
+    Hash,
+    PartialOrd,
+    Ord,
+)]
+#[repr(u32)] // it is u32 since used in Derivation Paths (CAP26) where each component is a u32.
+pub enum AbstractEntityType {
+    /// The entity type used by Accounts.
+    Account,
+    /// The entity type used by Personas.
+    Identity,
+    /// Resource address
+    Resource,
+}
+impl AbstractEntityType {
+    /// Conversion of the Radix Engines type for EntityType to Self.
+    pub fn try_from(value: EngineEntityType) -> Result<Self> {
+        match value {
+            EngineEntityType::GlobalVirtualEd25519Account => Ok(Self::Account),
+            EngineEntityType::GlobalVirtualSecp256k1Account => {
+                Ok(Self::Account)
+            }
+            EngineEntityType::GlobalVirtualEd25519Identity => {
+                Ok(Self::Identity)
+            }
+            EngineEntityType::GlobalFungibleResourceManager => {
```

Comment:
Either include `GlobalNonFungibleResourceManager` here, or make a specific `AbstractEntityType` for non fungible resources.

Reply:
I think I want a unit test first to see what it would give us, any and all code should be unit tested (and thus increase Code Coverage, not decrease it) - would it be for parsing of NonFungibleGlobalID? I have support for that using `EngineSerializableNonFungibleGlobalIdInternal::from_str` so it does not go through this method - we can change to that, but I'd prefer a focused PR for it :)

Reply:
Not only for NonFungibleGlobalID, It would be neat if we can use the Sargon's Address types in the Wallet also, where we do work more extensively with non fungible collections.

Reply:
Yes Sargon should uniffi::export the throwing FromStr constructors on all Address types (will be global freestanding functions due to Record method not supported - which can be made into true throwing units in Swift/Kotlin land using extensions)